src/deluge/gui/views/midi_session_view.cpp

```diff
@@ -0,0 +1,949 @@
+			                                              OLED_MAIN_WIDTH_PIXELS, kTextSpacingX, kTextSpacingY);
+		}
+		else {
+			deluge::hid::display::OLED::drawStringCentred(l10n::get(l10n::String::STRING_FOR_MIDI_NOT_LEARNED), yPos,
+			                                              deluge::hid::display::OLED::oledMainImage[0],
+			                                              OLED_MAIN_WIDTH_PIXELS, kTextSpacingX, kTextSpacingY);
+		}
+
+		deluge::hid::display::OLED::sendMainImage();
+	}
+	//7Seg Display
+	else {
+		if (ccNumber != kNoSelection) {
+			char buffer[5];
+			intToString(ccNumber, buffer);
+			display->displayPopup(buffer, 3, true);
+		}
+		else {
+			display->displayPopup(l10n::get(l10n::String::STRING_FOR_NONE), 3, true);
+		}
+	}
+	onParamDisplay = true;
+}
+
+void MidiSessionView::renderOLED(uint8_t image[][OLED_MAIN_WIDTH_PIXELS]) {
+	renderViewDisplay();
+}
+
+void MidiSessionView::redrawNumericDisplay() {
+	renderViewDisplay();
+}
+
+void MidiSessionView::setLedStates() {
+	setCentralLEDStates();  //inherited from session view
+	view.setLedStates();    //inherited from session view
+	view.setModLedStates(); //inherited from session view
+
+	//midi session view specific LED settings
+	indicator_leds::blinkLed(IndicatorLED::MIDI);
+	indicator_leds::blinkLed(IndicatorLED::LEARN);
+
+	if (currentSong->lastClipInstanceEnteredStartPos != -1) {
+		indicator_leds::blinkLed(IndicatorLED::SESSION_VIEW);
+	}
+}
+
+void MidiSessionView::setCentralLEDStates() {
+	indicator_leds::setLedState(IndicatorLED::SYNTH, false);
+	indicator_leds::setLedState(IndicatorLED::KIT, false);
+	indicator_leds::setLedState(IndicatorLED::CV, false);
+	indicator_leds::setLedState(IndicatorLED::SCALE_MODE, false);
+	indicator_leds::setLedState(IndicatorLED::CROSS_SCREEN_EDIT, false);
+	indicator_leds::setLedState(IndicatorLED::BACK, false);
+}
+
+ActionResult MidiSessionView::buttonAction(deluge::hid::Button b, bool on, bool inCardRoutine) {
+	using namespace deluge::hid::button;
+
+	//clear and reset learned params
+	if (b == BACK && isUIModeActive(UI_MODE_HOLDING_HORIZONTAL_ENCODER_BUTTON)) {
+		if (on) {
+			initPadPress(lastPadPress);
+			initMapping(paramToCC);
+			initMapping(previousKnobPos);
+			updateMappingChangeStatus();
+			uiNeedsRendering(this);
+		}
+	}
+
+	//save midi mappings
+	else if (b == SAVE) {
+		if (on) {
+			saveMidiFollowMappings();
+			display->displayPopup(l10n::get(l10n::String::STRING_FOR_MIDI_DEFAULTS_SAVED));
+		}
+	}
+
+	//load midi mappings
+	else if (b == LOAD) {
+		if (on) {
+			loadMidiFollowMappings();
+			display->displayPopup(l10n::get(l10n::String::STRING_FOR_MIDI_DEFAULTS_LOADED));
+		}
+	}
+
+	//enter "Midi Follow" soundEditor menu
+	else if ((b == SELECT_ENC) && !Buttons::isShiftButtonPressed()) {
+		if (on) {
+			display->setNextTransitionDirection(1);
+			soundEditor.setup();
+			openUI(&soundEditor);
+		}
+	}
+
+	//exit Midi View
+	else if (b == MIDI) {
```

Comment:
not just exiting, add description that it will highlight learned params also when learn is held